EDDiscovery/UserControls/UserControlStarDistance.cs

```diff
@@ -137,10 +138,44 @@ private void FillGrid(string name, SortedList<double, ISystem> csl)
                         }
                     }
                     else
-                    {
-                        int rowindex = dataGridViewNearest.Rows.Add(rowobj);
+                    {   int rowindex = dataGridViewNearest.Rows.Add(rowobj);
                         dataGridViewNearest.Rows[rowindex].Tag = tvp.Value;
                     }
+
+                    // Check for duplicated entries and remove them
+                    if (dataGridViewNearest.Rows.Count > 2)
+                    {
+                        for (int currentRow = 0; currentRow < dataGridViewNearest.Rows.Count; currentRow++)
+                        {
+                            var rowToCompare = dataGridViewNearest.Rows[currentRow]; // Get row to compare against other rows
+
+                            // Iterate through all rows 
+                            //
+                            foreach (DataGridViewRow row in dataGridViewNearest.Rows)
```

Comment:
I'm not sure how many rows we're expecting in this grid, but this inner loop only needs to look _forward_ from `currentRow + 1` inclusive.  Any rows between 0 and `currentRow` have already been checked against this row, and can't be a dupe.

This also simplifies some of the logic slightly by removing the need for the equality check.

Reply:
Thanks, @wokket . I'm currently traveling, so unable to work on that. If you have a working alternative, and if you are able to include it, it will be great!

Reply:
I haven't tried to update someone else's PR before, but I'll have a look today.... worst case I'll submit a change once this approved if you like.